build.gradle

```diff
@@ -310,13 +309,104 @@ subprojects {
+// from each package, but this can be configured by adjusting the publishLibrary variable.
+// Each subproject will place its artifacts into the central staging repository,
+// which will then be picked up by the bundle task
+if (Boolean.parseBoolean(publishBuild) && Boolean.parseBoolean(centralPublish)) {
+    var stagingRepo = layout.buildDirectory.dir('staging-repo')
+
+    // Task to create a bundle zip of all published artifacts
+    tasks.register('createMavenCentralBundle', Zip) {
+        description = "Creates a bundle zip of all artifacts for Maven Central upload"
+        group = "publishing"
+
+        dependsOn subprojects.collect { it.tasks.matching { it.name == 'publish' } }
+
+        archiveBaseName = "${rootProject.group}-${rootProject.name}"
+        archiveVersion = project.version
+        archiveClassifier = 'bundle'
+        destinationDirectory = layout.buildDirectory.dir('distributions')
+
+        from(stagingRepo) {
+            include "${rootProject.group.replaceAll('.', '/')}/**/${project.version}/*"
+            exclude '**/maven-metadata*'
+        }
+
+        includeEmptyDirs = false
+    }
+
+    // Task to upload the bundle to Maven Central
+    tasks.register('publishMavenCentralBundle') {
+        description = "Uploads the bundle zip to Maven Central Portal"
+        group = "publishing"
+
+        dependsOn tasks.named('createMavenCentralBundle')
+
+        doLast {
+            def sonatypeUsername = findProperty('sonatypeUsername')
+            def sonatypePassword = findProperty('sonatypePassword')
+
+            if (sonatypeUsername == null || sonatypePassword == null) {
+                throw new GradleException("sonatypeUsername and sonatypePassword properties must be set")
+            }
+
+            // Create base64 encoded credentials
+            def credentials = "${sonatypeUsername}:${sonatypePassword}"
+            def encodedCredentials = Base64.getEncoder().encodeToString(credentials.getBytes('UTF-8'))
+
+            // Get the bundle file
+            def bundleTask = tasks.named('createMavenCentralBundle').get()
+            def bundleFile = bundleTask.archiveFile.get().asFile
+
+            if (!bundleFile.exists()) {
+                throw new GradleException("Bundle file does not exist: ${bundleFile}")
+            }
+
+            logger.lifecycle("Uploading bundle: ${bundleFile.name} (${bundleFile.length() / 1024 / 1024} MB)")
+
+            // Upload using HttpURLConnection for multipart/form-data
+            def url = new URL('https://central.sonatype.com/api/v1/publisher/upload?publishingType=AUTOMATIC')
+            def connection = url.openConnection() as HttpURLConnection
+
+            try {
+                connection.setRequestMethod('POST')
+                connection.setDoOutput(true)
+                connection.setRequestProperty('Authorization', "Bearer ${encodedCredentials}")
+
+                def boundary = "----GradleBoundary${System.currentTimeMillis()}"
```

Comment:
Given that we are uploading a zip, would it be worthwhile to add a `uuid` other this to help ensure that the zip file itself doesn't have this sequence of bytes? (at least if I understand multipart correctly)
Or would it be worthwhile to validate on our end that the string we use for the boundary is not in the file?